### PR TITLE
Open USD account screen

### PR DESCRIPTION
**Pull request by kenanmamedoff**

### CODE REVIEW

File: brd-ios/breadwallet/Helpers/NonSelectableTextView.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    override public var selectedTextRange: UITextRange? {
+        get {
+            return nil
+        }
+        set { }
+    }
+    
+    override func point(inside point: CGPoint, with event: UIEvent?) -> Bool {
+        guard let pos = closestPosition(to: point),
+              let range = tokenizer.rangeEnclosingPosition(pos, with: .character, inDirection: .layout(.left)) else { return false }
```
Comment (dgacevic) [inline brd-ios/breadwallet/Helpers/NonSelectableTextView.swift:23]: else return formatting

File: brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountInteractor.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    func toggleTermsTickbox(viewAction: OpenAccountModels.TermsTickbox.ViewAction) {
+        dataStore?.termsTickbox = viewAction.value
+        
+        validate(viewAction: .init())
+    }
+    
+    func validate(viewAction: OpenAccountModels.Validate.ViewAction) {
+        let isESignTickboxValid = dataStore?.eSignTickbox == true
+        let isAgreementsTickboxValid = dataStore?.agreementsTickbox == true
+        let isTermsTickboxValid = dataStore?.termsTickbox == true
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountInteractor.swift:48]: Could probably just pass dataStore values in provide default as false, or?

Comment (dgacevic) [reply brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountInteractor.swift:48]: Or create `isValid` property on dataStore and move the logic there

Comment (kenanmamedoff) [reply brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountInteractor.swift:48]: I don't think it matters unless I am misunderstanding you

File: brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+            NSAttributedString.Key.backgroundColor: UIColor.clear,
+            NSAttributedString.Key.font: Fonts.Body.two]
+        
+        let partOne = NSMutableAttributedString(string: L10n.UsdAccount.Create.Benefit1.bold, attributes: partOneAttributes)
+        let partTwo = NSMutableAttributedString(string: " " + L10n.UsdAccount.Create.benefit1, attributes: partTwoAttributes)
+        let combined = NSMutableAttributedString()
+        combined.append(partOne)
+        combined.append(partTwo)
+        
+        let view = IconTitleViewView()
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift:86]: `IconTitleViewView`? Also, why is view being created in presenter

File: brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+        let partOne = NSMutableAttributedString(string: L10n.UsdAccount.Create.Benefit2.bold, attributes: partOneAttributes)
+        let partTwo = NSMutableAttributedString(string: " " + L10n.UsdAccount.Create.Benefit2.regular, attributes: partTwoAttributes)
+        let combined = NSMutableAttributedString()
+        combined.append(partOne)
+        combined.append(partTwo)
+        
+        let view = IconTitleViewView()
+        view.configure(with: .init())
+        view.setup(with: .init(icon: .image(Asset.dollar.image), title: .attributedText(combined)))
+        view.content.setupCustomMargins(all: .large)
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift:113]: Same here

File: brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+        let partOne = NSMutableAttributedString(string: L10n.UsdAccount.Create.Benefit3.bold, attributes: partOneAttributes)
+        let partTwo = NSMutableAttributedString(string: " " + L10n.UsdAccount.Create.Benefit3.regular, attributes: partTwoAttributes)
+        let combined = NSMutableAttributedString()
+        combined.append(partOne)
+        combined.append(partTwo)
+        
+        let view = IconTitleViewView()
+        view.configure(with: .init())
+        view.setup(with: .init(icon: .image(Asset.clock.image), title: .attributedText(combined)))
+        view.content.setupCustomMargins(all: .large)
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift:137]: Same

File: brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+        text.append(interactableText)
+        text.append(endText)
+        
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.alignment = NSTextAlignment.left
+        paragraphStyle.minimumLineHeight = 20
+        text.addAttribute(NSAttributedString.Key.paragraphStyle, value: paragraphStyle, range: NSRange(location: 0, length: text.length))
+        
+        let model = TickboxItemViewModel(title: .init(text: .attributedText(text, linkAttributes)))
+        return model
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift:183]: Can return on above line directly

File: brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+        
+        text.append(interactableText)
+        
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.alignment = NSTextAlignment.left
+        paragraphStyle.minimumLineHeight = 20
+        text.addAttribute(NSAttributedString.Key.paragraphStyle, value: paragraphStyle, range: NSRange(location: 0, length: text.length))
+        
+        let model = TickboxItemViewModel(title: .init(text: .attributedText(text, linkAttributes)))
+        return model
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/OpenAccount/OpenAccountPresenter.swift:294]: Same

File: brd-ios/breadwallet/src/Redesign/Views/Core/FETextView.swift [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+class FETextView: NonSelectableTextView, ViewProtocol {
+    var viewModel: FETextViewViewModel?
+    var config: FETextViewConfiguration?
+    var didTapUrl: ((URL?) -> Void)?
+    
+    func configure(with config: FETextViewConfiguration?) {
+        guard let config else { return }
+        
+        self.config = config
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Views/Core/FETextView.swift:57]: super.configure does that

File: brd-ios/breadwallet/src/Redesign/Views/Core/FETextView.swift [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        dataDetectorTypes = .all
+        backgroundColor = .clear
+        textContainer.lineFragmentPadding = 0
+        textContainerInset = .zero
+    }
+    
+    func setup(with viewModel: FETextViewViewModel?) {
+        guard let viewModel else { return }
+
+        self.viewModel = viewModel
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Views/Core/FETextView.swift:75]: super.setup does that

**Review by dgacevic — DISMISSED** [review]

**Review by dgacevic — DISMISSED** [review]

**Review by dgacevic — APPROVED** [review]

**Pull request merged**